lib/diary/diary.js

```diff
@@ -0,0 +1,147 @@
+      method: 'GET',
+      data: dateObject
+    })
+    .then((data) =>{
+      populateMealsTables(data)
+    })
+    .fail( (error) => {
+      console.error(error)
+    })
+  }
+
+  getPrevious() {
+    let todaysDate = new Date
+    todaysDate.setDate(todaysDate.getDate() + counter)
+
+    let formattedDate = todaysDate
+    let diaryDay = formattedDate.getDate()
+    let diaryMonth = formattedDate.getMonth() + 1
+    let diaryYear = formattedDate.getFullYear()
+
+    formattedDate = `${diaryYear}-0${diaryMonth}-0${diaryDay}`
+    todaysDate = todaysDate.toDateString().slice(4)
+    const dateObject = {date: formattedDate}
+    const date = `<div class='row'><div class='col s6 offset-s3' id='date-box'><button class='btn' id='prev-page-btn'><</button><h3 id='date-display'>${todaysDate}</h3><button class='btn' id='next-page-btn'>></button></div></div>`
+
+    $('#Lunch-body').empty()
+    $('#Dinner-body').empty()
+    $('#Snacks-body').empty()
+    $('#Breakfast-body').empty()
+
+    calories.resetCalories()
+
+    $('#date-container').empty()
+    $('#date-container').append(date)
+
+    return $.ajax({
+      url: API + 'diaries/meals',
+      method: 'GET',
+      data: dateObject
+    })
+    .then((data) => {
+      populateMealsTables(data)
+    })
+    .fail( (error) => {
+      console.error(error)
+    })
+  }
+
+  getNext() {
+    let todaysDate = new Date
+    todaysDate.setDate(todaysDate.getDate() + counter)
+
+    let formattedDate = todaysDate
+    let diaryDay = formattedDate.getDate()
+    let diaryMonth = formattedDate.getMonth() + 1
+    let diaryYear = formattedDate.getFullYear()
+
+    formattedDate = `${diaryYear}-0${diaryMonth}-0${diaryDay}`
+    todaysDate = todaysDate.toDateString().slice(4)
+    const dateObject = {date: formattedDate}
+    const date = `<div class='row'><div class='col s6 offset-s3' id='date-box'><button class='btn' id='prev-page-btn'><</button><h3 id='date-display' class='center'>${todaysDate}</h3><button class='btn' id='next-page-btn'>></button></div></div>`
+
+    $('#Lunch-body').empty()
+    $('#Dinner-body').empty()
+    $('#Snacks-body').empty()
+    $('#Breakfast-body').empty()
+
+    calories.resetCalories()
+
+    $('#date-container').empty()
+    $('#date-container').append(date)
+
+    return $.ajax({
+      url: API + 'diaries/meals',
+      method: 'GET',
+      data: dateObject
+    })
+    .then((data) => {
+      populateMealsTables(data)
+    })
+    .fail( (error) => {
+      console.error(error)
+    })
+  }
+}
+
+function populateMealsTables(data){
+  calories.resetCalories()
+  if (Array.isArray(data)) {
+    $('#date-box').attr('data-diary-id', `${data[0].diary_id}`)
+    data.forEach((meal) => {
+      let mealName = meal.name
+      foodAdder.addFoodstoMealTable(meal, mealName)
+    })
+  } else {
+    $('#date-box').attr('data-diary-id', `${data.id}`)
+  }
+}
+
+$(document).ready(function(){
+  const diary = new Diary
+
+  diaryFoodTable.getFoods()
+
```

Comment:
we can refactor this further and add a diary listeners function within the diary. We can then call it in the constructor.